### DISCUSSION TITLE
Not able to see download/upload data on Magma NMS UI

### DISCUSSION THREAD

**Discussion by shalinjshah18** [magma/magma#7938] [Q&A]
We have srsUE, srsEnodeB and Magma AGW setup on VM and Orchestrator on AWS Cloud.
While we run srsEnodeB and srsUE, UE is able to obtain IP address from AGW and can ping UE IP from AGW as well.

But when traffic using iperf is generated for upload/download it does not show up correctly on Magma NMS UI and only show traffic in kb.

![image (1)](https://user-images.githubusercontent.com/83823581/124962288-13d02600-e03c-11eb-82d7-6663e0e33936.png)

admin@Ubuntu-20:~$ iperf3 -c 192.168.128.18 -V
iperf 3.7
Linux Ubuntu-20 5.8.0-50-generic #56~20.04.1-Ubuntu SMP Mon Apr 12 21:46:35 UTC 2021 x86_64
Control connection MSS 32768
Time: Thu, 08 Jul 2021 17:00:30 GMT
Connecting to host 192.168.128.18, port 5201
      Cookie: 7v6yh6ekp73g2t5htdhhluq2vmjjku7ict5g
      TCP MSS: 32768 (default)
[  5] local 192.168.128.18 port 49860 connected to 192.168.128.18 port 5201
Starting Test: protocol: TCP, 1 streams, 131072 byte blocks, omitting 0 seconds, 10 second test, tos 0
[ ID] Interval           Transfer     Bitrate         Retr  Cwnd
[  5]   0.00-1.00   sec  2.37 GBytes  20.3 Gbits/sec    0   2.69 MBytes
[  5]   1.00-2.00   sec  3.02 GBytes  25.9 Gbits/sec    0   2.81 MBytes
[  5]   2.00-3.00   sec  2.94 GBytes  25.3 Gbits/sec    0   2.94 MBytes
[  5]   3.00-4.00   sec  2.61 GBytes  22.5 Gbits/sec    0   3.12 MBytes
[  5]   4.00-5.00   sec  2.69 GBytes  23.1 Gbits/sec    0   3.12 MBytes
[  5]   5.00-6.00   sec  2.80 GBytes  24.1 Gbits/sec    0   3.12 MBytes
[  5]   6.00-7.00   sec  2.78 GBytes  23.9 Gbits/sec    0   3.12 MBytes
[  5]   7.00-8.00   sec  2.63 GBytes  22.6 Gbits/sec    0   3.12 MBytes
[  5]   8.00-9.00   sec  2.61 GBytes  22.4 Gbits/sec    0   3.12 MBytes
[  5]   9.00-10.00  sec  2.66 GBytes  22.8 Gbits/sec    0   3.12 MBytes
- - - - - - - - - - - - - - - - - - - - - - - - -
Test Complete. Summary Results:
[ ID] Interval           Transfer     Bitrate         Retr
[  5]   0.00-10.00  sec  27.1 GBytes  23.3 Gbits/sec    0             sender
[  5]   0.00-10.00  sec  27.1 GBytes  23.3 Gbits/sec                  receiver
CPU Utilization: local/sender 65.1% (0.7%u/64.4%s), remote/receiver 0.6% (0.0%u/0.6%s)
snd_tcp_congestion cubic
rcv_tcp_congestion cubic

**Comment by pshelar** [comment]
@ardzoht can you have a look.

**Reply by shalinjshah18** [reply]
@ardzoht when I run iperf3 command on UE IP from AGW Machine I am getting below response on console.
192.168.128.38 is UE IP

admin@Ubuntu-20:~$ sudo iperf3 -c 192.168.128.38  -V
iperf 3.7
Linux Ubuntu-20 5.4.0-77-generic #86-Ubuntu SMP Thu Jun 17 02:35:03 UTC 2021 x86_64
Control connection MSS 1448
Time: Mon, 12 Jul 2021 17:26:46 GMT
Connecting to host 192.168.128.38, port 5201
      Cookie: bh2rdr5foa5vi25wkg3vls7p6xq7pr34mm7i
      TCP MSS: 1448 (default)
[  5] local 192.168.128.1 port 43752 connected to 192.168.128.38 port 5201
Starting Test: protocol: TCP, 1 streams, 131072 byte blocks, omitting 0 seconds, 10 second test, tos 0
[ ID] Interval           Transfer     Bitrate         Retr  Cwnd
[  5]   0.00-1.00   sec   107 KBytes   880 Kbits/sec    2   1.41 KBytes
[  5]   1.00-2.00   sec  0.00 Bytes  0.00 bits/sec    1   1.41 KBytes
[  5]   2.00-3.00   sec  0.00 Bytes  0.00 bits/sec    0   1.41 KBytes
[  5]   3.00-4.00   sec  0.00 Bytes  0.00 bits/sec    1   1.41 KBytes
[  5]   4.00-5.00   sec  0.00 Bytes  0.00 bits/sec    0   1.41 KBytes
[  5]   5.00-6.00   sec  0.00 Bytes  0.00 bits/sec    0   1.41 KBytes
[  5]   6.00-7.00   sec  0.00 Bytes  0.00 bits/sec    1   1.41 KBytes
[  5]   7.00-8.00   sec  0.00 Bytes  0.00 bits/sec    0   1.41 KBytes
[  5]   8.00-9.00   sec  0.00 Bytes  0.00 bits/sec    0   1.41 KBytes
[  5]   9.00-10.00  sec  0.00 Bytes  0.00 bits/sec    0   1.41 KBytes
- - - - - - - - - - - - - - - - - - - - - - - - -
Test Complete. Summary Results:
[ ID] Interval           Transfer     Bitrate         Retr
[  5]   0.00-10.00  sec   107 KBytes  88.0 Kbits/sec    5             sender
[  5]   0.00-10.00  sec  0.00 Bytes  0.00 bits/sec                  receiver
CPU Utilization: local/sender 0.0% (0.0%u/0.0%s), remote/receiver 0.0% (0.0%u/0.0%s)
snd_tcp_congestion cubic
rcv_tcp_congestion cubic

iperf Done.

**Reply by shalinjshah18** [reply]
@ardzoht when I run iperf3 command from Host machine on which srsUE and srsEnodeB is setup I am getting below response.


admin@Ubuntu-20:/tmp$ sudo iperf3 -c 192.168.128.38 -V
iperf 3.7
Linux Ubuntu-20 5.8.0-50-generic #56~20.04.1-Ubuntu SMP Mon Apr 12 21:46:35 UTC 2021 x86_64
Control connection MSS 32768
Time: Mon, 12 Jul 2021 17:28:47 GMT
Connecting to host 192.168.128.38, port 5201
      Cookie: wpaz3loisbswxtst7hefq57mscpdgf3xlchh
      TCP MSS: 32768 (default)
[  5] local 192.168.128.38 port 58554 connected to 192.168.128.38 port 5201
Starting Test: protocol: TCP, 1 streams, 131072 byte blocks, omitting 0 seconds, 10 second test, tos 0
[ ID] Interval           Transfer     Bitrate         Retr  Cwnd
[  5]   0.00-1.00   sec  3.83 GBytes  32.9 Gbits/sec    0   3.18 MBytes
[  5]   1.00-2.00   sec  3.70 GBytes  31.7 Gbits/sec    0   3.18 MBytes
[  5]   2.00-3.00   sec  3.27 GBytes  28.1 Gbits/sec    0   3.18 MBytes
[  5]   3.00-4.00   sec  3.71 GBytes  31.9 Gbits/sec    0   3.18 MBytes
[  5]   4.00-5.00   sec  3.25 GBytes  27.9 Gbits/sec    0   3.18 MBytes
[  5]   5.00-6.00   sec  3.37 GBytes  28.9 Gbits/sec    0   3.18 MBytes
[  5]   6.00-7.00   sec  3.84 GBytes  32.9 Gbits/sec    0   3.18 MBytes
[  5]   7.00-8.00   sec  3.69 GBytes  31.7 Gbits/sec    0   3.18 MBytes
[  5]   8.00-9.00   sec  3.59 GBytes  30.8 Gbits/sec    0   3.18 MBytes
[  5]   9.00-10.00  sec  3.60 GBytes  30.9 Gbits/sec    0   3.18 MBytes
- - - - - - - - - - - - - - - - - - - - - - - - -
Test Complete. Summary Results:
[ ID] Interval           Transfer     Bitrate         Retr
[  5]   0.00-10.00  sec  35.8 GBytes  30.8 Gbits/sec    0             sender
[  5]   0.00-10.00  sec  35.8 GBytes  30.8 Gbits/sec                  receiver
CPU Utilization: local/sender 82.6% (0.6%u/82.0%s), remote/receiver 5.6% (0.1%u/5.5%s)
snd_tcp_congestion cubic
rcv_tcp_congestion cubic

iperf Done.

**Reply by shalinjshah18** [reply]
@ardzoht 

Output of service303_cli.py metrics sessiond
[sessiond_metrics.txt](https://github.com/magma/magma/files/6803352/sessiond_metrics.txt)

Output of service303_cli.py metrics pipelined
[pipelined_metrics.txt](https://github.com/magma/magma/files/6803353/pipelined_metrics.txt)

**Reply by shalinjshah18** [reply]
@ardzoht 
Please find output of sudo ovs-ofctl dump-flows gtp_br0

admin@Ubuntu-20:~$ sudo ovs-ofctl dump-flows gtp_br0
[sudo] password for admin:
 cookie=0x0, duration=410733.326s, table=0, n_packets=266, n_bytes=24520, priority=10,in_port=ipfix0 actions=resubmit(,201)
 cookie=0x0, duration=70343.723s, table=0, n_packets=0, n_bytes=0, priority=10,tun_id=0xd,in_port="g_d834640a" actions=mod_dl_src:02:00:00:00:00:01,mod_dl_dst:ff:ff:ff:ff:ff:ff,load:0xd->NXM_NX_REG9[],load:0x759806608ad->OXM_OF_METADATA[],resubmit(,1)
 cookie=0x0, duration=410733.326s, table=0, n_packets=30973223, n_bytes=3032554543, priority=0 actions=resubmit(,1)
 cookie=0x0, duration=410729.360s, table=1, n_packets=2861, n_bytes=421836, priority=10,in_port=LOCAL actions=set_field:0x10->reg1,resubmit(,2),set_field:0->reg0,set_field:0->reg3
 cookie=0x0, duration=410729.360s, table=1, n_packets=31354985, n_bytes=3070140413, priority=10,in_port=mtr0 actions=set_field:0x10->reg1,resubmit(,2),set_field:0->reg0,set_field:0->reg3
 cookie=0x0, duration=410729.360s, table=1, n_packets=120, n_bytes=8400, priority=10,in_port="proxy_port" actions=set_field:0x10->reg1,resubmit(,2),set_field:0->reg0,set_field:0->reg3
 cookie=0x0, duration=410729.360s, table=1, n_packets=2076, n_bytes=9264, priority=0,reg6=0 actions=set_field:0x1->reg1,resubmit(,2),set_field:0->reg0,set_field:0->reg3
 cookie=0x0, duration=410729.360s, table=1, n_packets=0, n_bytes=0, priority=0,reg6=0x1 actions=set_field:0x10->reg1,resubmit(,2),set_field:0->reg0,set_field:0->reg3
 cookie=0x0, duration=410733.263s, table=2, n_packets=0, n_bytes=0, priority=12,arp,reg1=0x10,arp_tpa=10.1.0.1,arp_op=2 actions=set_field:0x1->reg6,resubmit(,3),set_field:0->reg0,set_field:0->reg3
 cookie=0x0, duration=410733.263s, table=2, n_packets=0, n_bytes=0, priority=12,arp,reg1=0x10,arp_tpa=10.1.0.1,arp_op=1 actions=move:NXM_OF_ETH_SRC[]->NXM_OF_ETH_DST[],mod_dl_src:8a:dc:26:1f:cf:3f,load:0x2->NXM_OF_ARP_OP[],move:NXM_NX_ARP_SHA[]->NXM_NX_ARP_THA[],load:0x8adc261fcf3f->NXM_NX_ARP_SHA[],move:NXM_OF_ARP_TPA[]->NXM_NX_REG0[],move:NXM_OF_ARP_SPA[]->NXM_OF_ARP_TPA[],move:NXM_NX_REG0[]->NXM_OF_ARP_SPA[],IN_PORT
 cookie=0x0, duration=410733.263s, table=2, n_packets=0, n_bytes=0, priority=12,ip,reg1=0x1,nw_dst=10.1.0.1 actions=set_field:8a:dc:26:1f:cf:3f->eth_dst,resubmit(,3),set_field:0->reg0,set_field:0->reg3
 cookie=0x0, duration=410733.263s, table=2, n_packets=0, n_bytes=0, priority=12,arp,reg1=0x10,arp_tpa=192.168.128.0/24,arp_op=2 actions=set_field:0x1->reg6,resubmit(,3),set_field:0->reg0,set_field:0->reg3
 cookie=0x0, duration=410733.263s, table=2, n_packets=47324, n_bytes=1987608, priority=12,arp,reg1=0x10,arp_tpa=192.168.128.0/24,arp_op=1 actions=move:NXM_OF_ETH_SRC[]->NXM_OF_ETH_DST[],mod_dl_src:b6:b5:a6:00:5a:4c,load:0x2->NXM_OF_ARP_OP[],move:NXM_NX_ARP_SHA[]->NXM_NX_ARP_THA[],load:0xb6b5a6005a4c->NXM_NX_ARP_SHA[],move:NXM_OF_ARP_TPA[]->NXM_NX_REG0[],move:NXM_OF_ARP_SPA[]->NXM_OF_ARP_TPA[],move:NXM_NX_REG0[]->NXM_OF_ARP_SPA[],IN_PORT
 cookie=0x0, duration=410733.263s, table=2, n_packets=0, n_bytes=0, priority=12,arp,reg1=0x1,arp_spa=192.168.128.0/24 actions=set_field:0x1->reg6,resubmit(,3),set_field:0->reg0,set_field:0->reg3
 cookie=0x0, duration=410733.263s, table=2, n_packets=12, n_bytes=1008, priority=10,ip,reg1=0x1 actions=set_field:b6:b5:a6:00:5a:4c->eth_dst,resubmit(,3),set_field:0->reg0,set_field:0->reg3
 cookie=0x0, duration=410733.263s, table=2, n_packets=0, n_bytes=0, priority=10,arp actions=drop
 cookie=0x0, duration=410733.263s, table=2, n_packets=31312706, n_bytes=3068591297, priority=0 actions=resubmit(,3),set_field:0->reg0,set_field:0->reg3
 cookie=0x0, duration=410733.263s, table=3, n_packets=31310642, n_bytes=3068583041, priority=0,reg1=0x10 actions=resubmit(,4),set_field:0->reg0,set_field:0->reg3
 cookie=0x0, duration=410733.263s, table=3, n_packets=2076, n_bytes=9264, priority=0,reg1=0x1 actions=resubmit(,21),set_field:0->reg0,set_field:0->reg3
 cookie=0x0, duration=410733.263s, table=4, n_packets=120, n_bytes=8400, priority=1,in_port="proxy_port" actions=drop
 cookie=0x0, duration=410733.263s, table=4, n_packets=31312598, n_bytes=3068583905, priority=0 actions=resubmit(,10),set_field:0->reg0,set_field:0->reg3
 cookie=0x0, duration=410729.359s, table=10, n_packets=0, n_bytes=0, priority=15,reg6=0x1 actions=resubmit(,20),set_field:0->reg0,set_field:0->reg3
 cookie=0x0, duration=410729.359s, table=10, n_packets=0, n_bytes=0, priority=12,ip,reg1=0x1,vlan_tci=0x0000/0x1000,nw_dst=10.1.0.1 actions=output:mtr0
 cookie=0x0, duration=410729.359s, table=10, n_packets=0, n_bytes=0, priority=12,ip,reg1=0x1,vlan_tci=0x1000/0x1000,nw_dst=10.1.0.1 actions=strip_vlan,output:mtr0
 cookie=0x0, duration=410729.359s, table=10, n_packets=31312598, n_bytes=3068583905, priority=10 actions=resubmit(,11),set_field:0->reg0,set_field:0->reg3
 cookie=0x0, duration=410729.355s, table=11, n_packets=31312598, n_bytes=3068583905, priority=0 actions=resubmit(,12),set_field:0->reg0,set_field:0->reg3
 cookie=0xfffffffffffffffe, duration=410729.358s, table=12, n_packets=30926366, n_bytes=3030585961, priority=0 actions=resubmit(,13),set_field:0->reg0,set_field:0->reg3
 cookie=0xfffffffffffffffe, duration=410729.355s, table=13, n_packets=30926366, n_bytes=3030585961, priority=0 actions=drop
 cookie=0x0, duration=410729.359s, table=20, n_packets=0, n_bytes=0, priority=12,reg10=0x1 actions=set_field:e6:8f:a2:80:80:80->eth_dst,output:"proxy_port"
 cookie=0x0, duration=410729.359s, table=20, n_packets=386220, n_bytes=37996936, priority=0,ip,reg1=0x10,vlan_tci=0x0000/0x1000 actions=output:NXM_NX_REG8[]
 cookie=0x0, duration=410729.359s, table=20, n_packets=0, n_bytes=0, priority=0,ip,reg1=0x10,vlan_tci=0x1000/0x1000 actions=strip_vlan,output:NXM_NX_REG8[]
 cookie=0x0, duration=410729.359s, table=20, n_packets=12, n_bytes=1008, priority=0,reg1=0x1 actions=LOCAL
 cookie=0x0, duration=410733.263s, table=21, n_packets=2076, n_bytes=9264, priority=0 actions=resubmit(,4),set_field:0->reg0,set_field:0->reg3

**Reply by ardzoht** [reply]
Thanks for sharing these @shalinjshah18 From the shared output the metric `ue_reported_usage_bytes` is reporting 138569 bytes for DL and 0 for UL, which indicates the values you are seeing on NMS. @koolzz does the flow rules indicate the same for what the metric is reporting?

**Reply by koolzz** [reply]
Does this have a policy rule setup for ue? If so what's the name

**Reply by shalinjshah18** [reply]
> Does this have a policy rule setup for ue? If so what's the name

Hi @koolzz earlier we did not have policy setup for this UE. But now we did setup policy after guidance of Adam. And enabled debug logs on sessiond as well.

please find below output from sessiond service after setting up policy.

Jul 14 12:55:47 Ubuntu-20 sessiond[2147608]: I0714 12:55:47.095798 2147616 LocalEnforcer.cpp:1880] Pipelined add ue enf flow succeeded for IMSI001010123456789
Jul 14 12:55:49 Ubuntu-20 sessiond[2147608]: I0714 12:55:49.080246 2147608 SpgwServiceClient.cpp:121] Creating dedicated bearers for IMSI001010123456789, 192.168.128.42, rules={ NO_APP_NAME }
Jul 14 12:55:49 Ubuntu-20 sessiond[2147608]: I0714 12:55:49.083900 2147785 LocalSessionManagerHandler.cpp:563] Received a BindPolicy2Bearer request for IMSI001010123456789 with default bearerID: 9 policyID: NO_APP_NAME created dedicated bearerID: 0
Jul 14 12:55:49 Ubuntu-20 sessiond[2147608]: I0714 12:55:49.084522 2147608 LocalEnforcer.cpp:2071] Removing NO_APP_NAME since we failed to create a dedicated bearer for it
Jul 14 12:55:49 Ubuntu-20 sessiond[2147608]: I0714 12:55:49.084547 2147608 PipelinedClient.cpp:271] Deactivating 1 rules and for subscriber IMSI001010123456789 IP 192.168.128.42
Jul 14 12:55:49 Ubuntu-20 sessiond[2147608]: I0714 12:55:49.084614 2147608 LocalSessionManagerHandler.cpp:575] Failed to process policy -> bearer binding for NO_APP_NAME
Jul 14 12:55:49 Ubuntu-20 sessiond[2147608]: I0714 12:55:49.085520 2147608 LocalSessionManagerHandler.cpp:580] Succeeded in updating SessionStore after processing policy->bearer mapping
Jul 14 12:55:51 Ubuntu-20 sessiond[2147608]: I0714 12:55:51.565452 2147608 LocalSessionManagerHandler.cpp:66] Aggregating 2 records
Jul 14 12:55:51 Ubuntu-20 sessiond[2147608]: I0714 12:55:51.565482 2147608 SessionState.cpp:468] Updating used charging credit for Rule=NO_APP_NAME Rating Group=1 Service Identifier=0
Jul 14 12:55:51 Ubuntu-20 sessiond[2147608]: I0714 12:55:51.565490 2147608 SessionCredit.cpp:622] ===> Used     Tx: 0 Rx: 190655 Total: 190655
Jul 14 12:55:51 Ubuntu-20 sessiond[2147608]: I0714 12:55:51.565493 2147608 SessionCredit.cpp:625] ===> Reported Tx: 0 Rx: 0 Total: 0
Jul 14 12:55:51 Ubuntu-20 sessiond[2147608]: I0714 12:55:51.565496 2147608 SessionCredit.cpp:628] ===> Allowed  Tx: 0 Rx: 0 Total: 0
Jul 14 12:55:51 Ubuntu-20 sessiond[2147608]: I0714 12:55:51.565500 2147608 SessionCredit.cpp:631] ===> A_Floor  Tx: 0 Rx: 0 Total: 0
Jul 14 12:55:51 Ubuntu-20 sessiond[2147608]: I0714 12:55:51.565503 2147608 SessionCredit.cpp:634] ===> (%used)  Tx: _% Rx: _% Total: _%
Jul 14 12:55:51 Ubuntu-20 sessiond[2147608]: I0714 12:55:51.565506 2147608 SessionCredit.cpp:647] ===> Grant tracking type TRACKING_UNSET,  Reporting: 0
Jul 14 12:55:51 Ubuntu-20 sessiond[2147608]: I0714 12:55:51.565510 2147608 SessionCredit.cpp:652] ===> Last Granted Units Received (tx/rx/total) 0/0/0
Jul 14 12:55:51 Ubuntu-20 sessiond[2147608]: I0714 12:55:51.565538 2147608 SessionState.cpp:1171] Session IMSI001010123456789-152268 Teid 0 FSM state change from SESSION_RELEASED to SESSION_TERMINATED
Jul 14 12:55:51 Ubuntu-20 sessiond[2147608]: I0714 12:55:51.565546 2147608 SessionCredit.cpp:453] ===> Data usage since last report is tx=0 rx=190655
Jul 14 12:55:51 Ubuntu-20 sessiond[2147608]: I0714 12:55:51.565549 2147608 SessionCredit.cpp:677] ===> Amount reporting for this report: tx=0 rx=190655
Jul 14 12:55:51 Ubuntu-20 sessiond[2147608]: I0714 12:55:51.565552 2147608 SessionCredit.cpp:679] ===> The total amount currently being reported: tx=0 rx=190655
Jul 14 12:55:51 Ubuntu-20 sessiond[2147608]: I0714 12:55:51.565745 2147608 LocalEnforcer.cpp:1153] IMSI001010123456789-152268 deleted from SessionMap
Jul 14 12:55:51 Ubuntu-20 sessiond[2147608]: I0714 12:55:51.568029 2147608 LocalSessionManagerHandler.cpp:92] Succeeded in updating session after no reporting
Jul 14 12:55:51 Ubuntu-20 sessiond[2147608]: I0714 12:55:51.568073 2147608 SessionReporter.cpp:45] Termination successful in controller for IMSI001010123456789-152268
Jul 14 12:55:51 Ubuntu-20 sessiond[2147608]: I0714 12:55:51.617101 2147612 PolicyLoader.cpp:70] 1 rules synced
Jul 14 12:55:51 Ubuntu-20 sessiond[2147608]: I0714 12:55:51.769220 2147608 LocalEnforcer.cpp:420] Forced termination timeout! Checking if termination has to be forced for IMSI001010123456789-152268... => NO
Jul 14 12:56:01 Ubuntu-20 sessiond[2147608]: I0714 12:56:01.320866 2147785 LocalSessionManagerHandler.cpp:597] Received a UpdateTunnelIds request for IMSI001010123456789 with default bearer id: 9 enb_teid= 8 agw_teid= 43
Jul 14 12:56:01 Ubuntu-20 sessiond[2147608]: I0714 12:56:01.321313 2147608 PipelinedClient.cpp:297] Activating 0 rules for IMSI001010123456789 msisdn  and ip 192.168.128.42
Jul 14 12:56:01 Ubuntu-20 sessiond[2147608]: I0714 12:56:01.321394 2147608 LocalEnforcer.cpp:1984] Updating IPFIX flow for subscriber IMSI001010123456789
Jul 14 12:56:01 Ubuntu-20 sessiond[2147608]: I0714 12:56:01.321401 2147608 LocalEnforcer.cpp:1990] Failed mac/name parsiong for apn internet
Jul 14 12:56:01 Ubuntu-20 sessiond[2147608]: I0714 12:56:01.325122 2147616 LocalEnforcer.cpp:1880] Pipelined add ue enf flow succeeded for IMSI001010123456789
Jul 14 12:56:01 Ubuntu-20 sessiond[2147608]: I0714 12:56:01.566131 2147608 LocalSessionManagerHandler.cpp:66] Aggregating 1 records
Jul 14 12:56:01 Ubuntu-20 sessiond[2147608]: I0714 12:56:01.567015 2147608 LocalSessionManagerHandler.cpp:92] Succeeded in updating session after no reporting
Jul 14 12:56:06 Ubuntu-20 sessiond[2147608]: I0714 12:56:06.617406 2147612 PolicyLoader.cpp:70] 1 rules synced
Jul 14 12:56:11 Ubuntu-20 sessiond[2147608]: I0714 12:56:11.566956 2147608 LocalSessionManagerHandler.cpp:66] Aggregating 1 records
Jul 14 12:56:11 Ubuntu-20 sessiond[2147608]: I0714 12:56:11.568110 2147608 LocalSessionManagerHandler.cpp:92] Succeeded in updating session after no reporting
Jul 14 12:56:21 Ubuntu-20 sessiond[2147608]: I0714 12:56:21.570062 2147608 LocalSessionManagerHandler.cpp:66] Aggregating 1 records
Jul 14 12:56:21 Ubuntu-20 sessiond[2147608]: I0714 12:56:21.570763 2147608 LocalSessionManagerHandler.cpp:92] Succeeded in updating session after no reporting
Jul 14 12:56:21 Ubuntu-20 sessiond[2147608]: I0714 12:56:21.617775 2147612 PolicyLoader.cpp:70] 1 rules synced
Jul 14 12:56:31 Ubuntu-20 sessiond[2147608]: I0714 12:56:31.571709 2147608 LocalSessionManagerHandler.cpp:66] Aggregating 1 records
Jul 14 12:56:31 Ubuntu-20 sessiond[2147608]: I0714 12:56:31.572512 2147608 LocalSessionManagerHandler.cpp:92] Succeeded in updating session after no reporting
Jul 14 12:56:36 Ubuntu-20 sessiond[2147608]: I0714 12:56:36.618113 2147612 PolicyLoader.cpp:70] 1 rules synced
Jul 14 12:56:41 Ubuntu-20 sessiond[2147608]: I0714 12:56:41.573506 2147608 LocalSessionManagerHandler.cpp:66] Aggregating 1 records
Jul 14 12:56:41 Ubuntu-20 sessiond[2147608]: I0714 12:56:41.574782 2147608 LocalSessionManagerHandler.cpp:92] Succeeded in updating session after no reporting

**Reply by ulaskozat** [reply]
what do you use as APN AMBR? 

Also why do you set UE policy? And what did you set? Is your RAN simulator capable of handling dedicated bearers?

**Reply by shalinjshah18** [reply]
[Policy-JSON.txt](https://github.com/magma/magma/files/6817619/Policy-JSON.txt)
![Policy Configuration 1](https://user-images.githubusercontent.com/83823581/125659385-90bf9b67-1247-4d47-b4a0-dbf448041b40.PNG)
![APN Config in Magma NMS](https://user-images.githubusercontent.com/83823581/125659390-c9389d2a-1b5f-41ae-ba27-dd3e28f4b069.png)

@ulaskozat Please find 3 attachments; APN Config in Magma NMS.png is APN configuration performed in Magma NMS UI.
Policy Configuration 1.PNG is screenshot of Policy Name; 

Policy-JSON.txt contains JSON output of Policy configuration done inside Magma UI and All Flow rules defined.

[Subscriber-JSON.txt](https://github.com/magma/magma/files/6817642/Subscriber-JSON.txt)
Subscriber-JSON.txt is file where it both APN and Policy are attached to Subscriber.

We create Policy since Adam suggested to start debug logs on sessiond and looks at Tx, Rx info. This was not present before policy was created and than attached to Subscriber inside sessiond logs. Hence, we created Policy;

we are using srsUE and srsEnodeb simulators; I am not sure if they are capable of handling dedicated bearers of not. But they both simulators are able to connect to Magam AGW and able to get UE IP.
